FinSearchUnified/tests/Helper/StaticHelperTest.php

```diff
@@ -152,4 +190,29 @@ function testUseShopSearch(
         $shop = $expectedResult ? "shop" : "FINDOLOGIC";
         $this->assertEquals($result, $expectedResult, sprintf($error, $shop));
     }
+
+    /**
+     * @dataProvider controlCharacterProvider
+     *
+     * @param string $text
+     * @param string $expected
+     */
+    public function testControlCharacterMethod($text, $expected)
+    {
+        $result = StaticHelper::removeControlCharacters($text);
+        $this->assertEquals($expected, $result, sprintf('%s was expected but %s was returned', $expected, $result));
+    }
+
+    /**
+     * @dataProvider cleanStringProvider
+     * @dataProvider controlCharacterProvider
+     *
+     * @param string $text
+     * @param string $expected
+     */
+    public function testCleanStringMethod($text, $expected)
+    {
+        $result = StaticHelper::cleanString($text);
+        $this->assertEquals($expected, $result, sprintf('%s was expected but %s was returned', $expected, $result));
```

Comment:
The default assertion message will actually say the same as the one you provided. I would suggest adding the messages to the data provider and being more specific e.g.
- Expected control characters to be stripped way.
- Expected umlauts to be left unaltered.

and so on.